lib/parse.js

```diff
@@ -811,18 +833,49 @@ define(['esprima', './lang'], function (esprima, lang) {
     };
 
 
-    parse.findRequireDepNames = function (node, deps) {
+    parse.findRequireDepNames = function (node, deps, params) {
+        var param;
         traverse(node, function (node) {
             var arg;
 
-            if (node && node.type === 'CallExpression' && node.callee &&
-                    node.callee.type === 'Identifier' &&
-                    node.callee.name === 'require' &&
-                    node[argPropName] && node[argPropName].length === 1) {
+            if (node) {
+                if (node.type === 'CallExpression' && node.callee &&
+                        node.callee.type === 'Identifier' &&
+                        node.callee.name === 'require' &&
+                        node[argPropName] && node[argPropName].length === 1) {
+
+                    arg = node[argPropName][0];
+                    if (arg.type === 'Literal') {
+                        deps.push(arg.value);
+                        if (param) {
+                            params.push(param);
+                        }
+                    }
+                } else if (node.type === 'VariableDeclarator') {
+                    if (node.id && node.id.type === 'Identifier') {
+                        param = node.id.name;
+                    }
+                    return;
+                } else if (node.type === 'AssignmentExpression') {
+                    if (node.left && node.left.type === 'Identifier') {
+                        param = node.left.name;
+                    }
+                    return;
+                } else if (node.type === 'Property') {
+                    if (node.key && node.key.type === 'Identifier') {
+                        param = node.key.name;
+                    }
+                    return;
+                } else if (node.type === 'Identifier') {
+                    //Do not clear the parameter name in the current
+                    //declaration or assignment. The identifier is parsed
+                    //between the left and right parts of the statement.
+                    return;
+                }
 
-                arg = node[argPropName][0];
-                if (arg.type === 'Literal') {
-                    deps.push(arg.value);
+                // If range or loc options are set, nodes without type occur.
+                if (node.type) {
```

Comment:
So this is saying that if the node does have a type that the `param` should be cleared? On first read of the comment above this line, it gave me to impression that `if (!node.type)` was the interesting case?

Reply:
On the first impression, this condition shouldn't be needed. Previous conditions end with `return`. If the code runs to the `param = null`, it should be an unexpected expression and it wouldn't make sense to retain the current parsing state (the parameter name).

It is true with the default Esprima parsing options. However, if `loc` or `range` are enabled, the AST contain nodes without type - whitespace, for example. That is why I need to ignore not only visitor calls, where `node == null`, but also where `node.type == null`. I could've changed the [line 841 in `findRequireDepNames`](https://github.com/amodrojs/amodro-trace/pull/14/files#diff-9f97a9c36fbbfb0d19068d6fae3bb84eR841) from:

```js
if (node) {
```

to:

```js
if (node && node.type) {
```

Not to need that condition and have just `param = null`.